### PR TITLE
backoff retry + added delay to weekly agg updates

### PR DESCRIPTION
**Pull request by Sarahtonein**
Noticed that connections were timing out with a 502 Gateway error.

Added backoff retry.

Noticed agg updates were running everytime bot was initiated.

Added initial 7 day delay from time bot is started.

### CODE REVIEW

**Comment by Balugrizzly** [comment]
If the bot gets restarted twice on the sixth day 
would the next update take 19 days in this case?
If thats the case changing that down the road would be a good thing

File: notion.py [hunk 1]
```diff
@@ -295,4 +310,4 @@ async def handle_aggregate_updates(chan):
 
 
 if __name__ == "__main__":
-    asyncio.run(handle_updates(1130461410871226368))
+    asyncio.run(handle_updates(1130461410871226368))
```
Comment (Balugrizzly) [inline notion.py:313]: Ty for changes

**Review by Balugrizzly — APPROVED** [review]
@Sarahtonein 
Looks good 
coordinate with me for the minimal downtime deployment

**Comment by Sarahtonein** [comment]
> If the bot gets restarted twice on the sixth day would the next update take 19 days in this case? If thats the case changing that down the road would be a good thing

Good question, I don't think that is the case, I think that it would restart on the second restart and wait 7 days from that point in time.

**Comment by Balugrizzly** [comment]
Did you mean perform a agg update on the second restart? if thats the case the same would apply in the other direction aka daily updates would result from daily restarts which is also not so ideal and smthing for down the road 

otherwise it does exactly what i said

**Comment by Sarahtonein** [comment]
> Did you mean perform a agg update on the second restart? if thats the case the same would apply in the other direction aka daily updates would result from daily restarts which is also not so ideal and smthing for down the road
> 
> otherwise it does exactly what i said

I get your initial question now, and the answer would be yes.

I think what I should have done would be to keep track of the date/time the bot was started, and to check that when doing the weekly updates. This way it is not reset if the bot is restarted.

**Comment by Balugrizzly** [comment]
Yes that would be a valid approach 
If you don't take this on immediately make a issue and document it so its not forgotten

**Comment by Sarahtonein** [comment]
> Yes that would be a valid approach If you don't take this on immediately make a issue and document it so its not forgotten

I'll make an issue and circle back to this post Discord bot work

**Pull request merged**